app/src/main/java/at/bitfire/icsdroid/db/DatabaseAndroidInterface.kt

```diff
@@ -24,22 +24,22 @@ class DatabaseAndroidInterface(
     private val context: Context,
     private val subscription: Subscription,
 ) {
-    /**
-     * Gets the calendar provider for a given context.
-     * @param context The context that is making the request.
-     * @return The [ContentProviderClient] that provides an interface with the system's calendar.
-     */
-    private fun getProvider(context: Context) =
-        context.contentResolver.acquireContentProviderClient(CalendarContract.AUTHORITY)
+    companion object {
+        /**
+         * Gets the calendar provider for a given context.
+         * @return The [ContentProviderClient] that provides an interface with the system's calendar.
+         */
+        fun getProvider(context: Context) =
+            context.contentResolver.acquireContentProviderClient(CalendarContract.AUTHORITY)
```

Comment:
It should be clear whether this may return `null` or not, and what happens when calendar permissions are not given (`acquireContentProviderClient` throws `SecurityException`).

As far as I remember, `acquireContentProviderClient` can in general
- return `null` for instance if the calendar provider is not available (maybe some rare Android TVs or something like that, or when the user explicitly disables the system component, can happen for instance with custom ROMs)
- throw `SecurityException` when permissions are not granted

But please check whether I remember correctly.

Anyway, those cases should be handled and/or documented in KDoc.

Reply:
The [Android reference](https://developer.android.com/reference/kotlin/android/content/ContentResolver#acquirecontentproviderclient) doesn't say anything about throws, but we know they are not always annotated.

However, I have found a snippet:

https://github.com/sephiroth74/ImageViewZoom/blob/bfe833a7c61bcc28fea39f5758f0531cde0f4dd8/demo/src/it/sephiroth/android/library/imagezoom/test/utils/ExifUtils.java#L150-L156

that does indeed handle `SecurityException` so I guess you are right. I will annotate them.